share/tunnel/tunnel.go

```diff
@@ -68,6 +68,10 @@ func New(c Config) *Tunnel {
 	return t
 }
 
+func (t *Tunnel) Close() {
+	t.activatingConn.DoneAll()
```

Comment:
activatingConn is a waitGroup, it is value type and thread safe, so no need for lock and nil checks

Reply:
ah yea that's right, its a custom wait group (which might be a bad idea itself - anyway)

it should block while disconnected/connecting  and it should not-block while connected

DoneAll here would stop blocking, but Tunnel Done means closed?

how does this improve UDP? sorry, I think I'm missing something

Reply:
In the original code, when the connection is lost, DoneAll in the BindSSH method will be called, which causes activatingConnWait() to not work and cannot block.

This leads to high resource usage of runOutbound in udpListener. Calling `u.getUDPChan(ctx)` will immediately gets nil instead of blocking.

`Close` is added to correctly release resources in server mode, because DoneAll was removed from BindSSH.

In client mode, there is no need to call DoneAll because the same tunnel is always used.